fixes #18826: containers do not restart after reboot…

Dmitry-k42 opened the pull request:
…when bound to virtual network interface

Waiting for network is up when using systemd as described in [https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/](https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/).

In contrib/init/systemd/docker.service* string
`After=network.target docker.socket`
replaced to
`After=network-online.target docker.socket`
Probably fix for issue #18826

yongtang committed:
Support multiple service IDs on "docker service ps"

This fix tries to address issue raised in 25228 to support
multiple service IDs on `docker service ps`.

Multiple IDs are allowed with `docker service ps ...`, and
related documentation has been updated.

A test has been added to cover the changes.

This fix fixes 25228.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

dnephin committed:
Read long description from a file.

Signed-off-by: Daniel Nephin <dnephin@docker.com>

allencloud committed:
split function out of command description scope

Signed-off-by: allencloud <allen.sun@daocloud.io>

Commit:
Rename 'context' to 'loginfo' in the logger module

Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

gofmt

Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

update

Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

remove 'api/types/container/config.go' from this PR

Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

change 'LogInfo' to 'Info'

Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

update

Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

vdemeester committed:
Clean integration-cli/utils.go from most of its content

Most of the code is now on pkg/integration.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

Commit:
Update check-config.sh to include modules for FTP/TFTP

Signed-off-by: Santhosh Manohar <santhosh@docker.com>

vieux committed:
replace no-remove by sample-volume-plugin in docs

Signed-off-by: Victor Vieux <vieux@docker.com>

AkihiroSuda committed:
vendor oauth2@96382aa0 and add its deps

Signed-off-by: Akihiro Suda <suda.akihiro@lab.ntt.co.jp>

AkihiroSuda committed:
vendor: google.golang.org/cloud -> cloud.google.com/go

Signed-off-by: Akihiro Suda <suda.akihiro@lab.ntt.co.jp>

vdemeester committed:
Add the possibility to set the dev container name

By using `make CONTAINER_NAME=docker-dev shell` you're able to name
the docker-dev container to what you want. The idea behind it is by
knowing the name of your dev container you can automate easily some
thing using `docker exec` later.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

AkihiroSuda committed:
gcplogs: forcibly set HOME on static UNIX binary

Fix #29344

If HOME is not set, the gcplogs logging driver will call os/user.Current() via oauth2/google.
However, in static binary, os/user.Current() leads to segfault due to a glibc issue that won't be fixed
in a short term. (golang/go#13470, https://sourceware.org/bugzilla/show_bug.cgi?id=19341)
So we forcibly set HOME so as to avoid call to os/user/Current().

Signed-off-by: Akihiro Suda <suda.akihiro@lab.ntt.co.jp>

mavenugo committed:
Handle Plugin reference count during network create and delete

Signed-off-by: Madhu Venugopal <madhu@docker.com>

thaJeztah committed:
Update criu to 2.9

criu 2.8 and 2.9 contain various fixes,
so updating the version in the Dockerfile
to match the newer version.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

mavenugo committed:
Vendoring libnetwork to bring in isbuiltin changes

Signed-off-by: Madhu Venugopal <madhu@docker.com>

dnephin committed:
Replace vendor of aanand/compose-file with a local copy.

Add go-bindata for including the schema.

Signed-off-by: Daniel Nephin <dnephin@docker.com>

dnephin committed:
Add deps to test compose-file.

Signed-off-by: Daniel Nephin <dnephin@docker.com>

yongtang committed:
Sort the build labels passed from `build --labels`

This fix tries to fix the issue in 29619 where
labels passed from `build --labels` are not sorted.
As a result, if multiple labels have been passed,
each `docker build --labels A=A --labels B=B --labels C=C`
will generate different layers.

This fix fixes the issue by sort the Labels before
they are concatenated to `LABEL ...`.

A unit test has been added to cover the changes

This fix fixes 29619.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

allencloud committed:
support registry mirror config reload

Signed-off-by: allencloud <allen.sun@daocloud.io>

mavenugo committed:
Properly cleanup plugin states which might impact other tests

Signed-off-by: Madhu Venugopal <madhu@docker.com>

tonistiigi committed:
Define PushResult in api types

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

tonistiigi committed:
Move builder cli helper functions to own pkg

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

tonistiigi committed:
Move imageID validation to stringid pkg

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

tonistiigi committed:
Move UAStringKey to dockerversion pkg

Removes grpc dependency from client

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

yongtang committed:
Tests to allow ID-based `docker plugin enable/disable/rm/set`
This fix is a follow up based on comment:

and a follow up to:
https://github.com/docker/docker/pull/29222#issuecomment-268908937

As #28789 has been merged in, it is possible for `docker plugin inspect`
to search based on Name or ID Prefix. However, ID-based
`docker plugin enable/disable/rm/set` are still not possible.

This fix addes test for  `docker plugin enable/disable/rm/set` to search based on:
- Full ID
- Full Name
- Partial ID (prefix)

The actual fix is done in  #29487.

This fix is a follow up of #28789 and  #29487.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

allencloud committed:
fix nits in comments

Signed-off-by: allencloud <allen.sun@daocloud.io>

vdemeester committed:
Clean some stuff from runconfig that are cli only…

… or could be in `opts` package. Having `runconfig/opts` and `opts`
doesn't really make sense and make it difficult to know where to put
some code.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

yongtang committed:
Fix image's `CMD` after `WORKDIR` in Dockerfile

This fix tries to fix 29667 where image's `CMD` is modified
after `WORKDIR` in Dockerfile.

The value of `b.runConfig.Cmd` was modified in the processing
of `WORKDIR`, in order to fix 28902. However, the same
`b.runConfig.Cmd` is passed to `commit()`.

This fix restored the `b.runConfig.Cmd` before `commit()`
the image for `WORKDIR`.

A test has been added.

This fix fixes 29667.

This fix is related to 28902, 28909, 28514.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

vdemeester committed:
Enhance pkg/{httputils,integration}, distribution/xfer unit tests

- Make it possible to define a shorter waiting time of httputils
- Make a small hack to reduce the waiting time on distribution/xfer

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

thaJeztah committed:
Minor cleanups in cli/command/container

This change does some minor cleanups in the
cli/command/container package;

- sort imports
- replace `fmt.Fprintf()` with `fmt.Fprintln()` if no formatting is used
- replace `fmt.Errorf()` with `errors.New()` if no formatting is used
- remove some redundant `else`'s

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

Commit:
Remove redundant format

Signed-off-by: Ke Li <kel@splunk.com>

Add missing changes

Signed-off-by: Ke Li <kel@splunk.com>

User errors.New to create error

Signed-off-by: Ke Li <kel@splunk.com>

vdemeester committed:
Introduce a environment package in integration-cli

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

vdemeester committed:
Move package cliconfig to cli/config

I felt it made more sence 👼

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

lixiaobing1 committed:
It is errNoSuchWatch not errNoSuchPoller in filenotify/poller.go

Signed-off-by: lixiaobing10051267 <li.xiaobing1@zte.com.cn>

zhenghenghuo committed:
fix some typos

Signed-off-by: zhenghenghuo <zhenghenghuo@zju.edu.cn>

try to pass test

Signed-off-by: zhenghenghuo <zhenghenghuo@zju.edu.cn>

try to pass the test

Signed-off-by: zhenghenghuo <zhenghenghuo@zju.edu.cn>

yuexiao-wang committed:
Update docker daemon to dockerd

Signed-off-by: yuexiao-wang <wang.yuexiao@zte.com.cn>

yuexiao-wang committed:
Update 'ID' field for 'docker plugin ls'

Signed-off-by: yuexiao-wang <wang.yuexiao@zte.com.cn>

vdemeester committed:
Display only the name of the requirement…

… relative to the integration-cli package

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

yongtang committed:
Remove `docker stack ps -a` to match removal of `docker service/node ps -a`

In #28507 and #28885, `docker service/node ps -a` has been removed so that
information about slots are show up even without `-a` flag.

The output of `docker stack ps` reused the same output as `docker service/node ps`.
However, the `-a` was still there. It might make sense to remove `docker stack ps -a`
as well to bring consistency with `docker service/node ps`.

This fix is related to #28507, #28885, and #25983.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

lixiaobing1 committed:
use https clone url for docker/docker in ARM.md
Signed-off-by: lixiaobing10051267 <li.xiaobing1@zte.com.cn>

Dmitry-k42 committed:
bugfix issie 18826: containers do not restart after reboot when bound to virtual network interface

Signed-off-by: Dmitry Kononenko <d@dm42.ru>

GordonTheTurtle commented:
<!-- AUTOMATED:POULE:DCO-EXPLANATION -->
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "fix_issue_18826" git@github.com:mentalMedley/docker.git somewhere
$ cd somewhere
$ git commit --amend -s --no-edit
$ git push -f
~~~

Amending updates the existing PR. You **DO NOT** need to open a new one.

thaJeztah commented:
Thanks!

ping @andrewhsu PTAL

cpuguy83 committed:
Fix race/deadlock in v1 plugin handlers

When a plugin is activated, and then `plugins.Handle` is called to
register a new handler for a given plugin type, a deadlock occurs when
for anything which calls `waitActive`, including `Get`, and `GetAll`.

This happens because `Handle()` is setting `activated` to `false` to
ensure that plugin handlers are run on next activation.
Maybe these handlers should be called immediately for any plugins which
are already registered... but to preserve the existing behavior while
fixing the deadlock, track if handlers have been run on plugins and
reset when a new handler is registered.

The simplest way to reproduce the deadlock with Docker is to add a `-v
/foo` to the test container created for the external graphdriver tests.

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

dnephin committed:
Add validation for compose schema bindata.

Signed-off-by: Daniel Nephin <dnephin@docker.com>

GordonTheTurtle commented:
<!-- AUTOMATED:POULE:DCO-EXPLANATION -->
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "fix_issue_18826" git@github.com:mentalMedley/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~842354521360
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Amending updates the existing PR. You **DO NOT** need to open a new one.

yuexiao-wang committed:
Remove unused variable

Signed-off-by: yuexiao-wang <wang.yuexiao@zte.com.cn>

dmcgowan committed:
Support for docker content trust for plugins

Add integration test for docker content trust

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

tonistiigi committed:
Fix validation of plugins without rootfs in config

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

tonistiigi committed:
Add test TestRestartPolicyWithLiveRestore

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

lixiaobing1 committed:
update URL of plugins_graphdriver.md in experimental/README.md
Signed-off-by: lixiaobing10051267 <li.xiaobing1@zte.com.cn>

yuexiao-wang committed:
keep network option consistent between network connect and run

Signed-off-by: yuexiao-wang <wang.yuexiao@zte.com.cn>

yuexiao-wang committed:
Update the field 'CONTAINER ID' for docker ps

Signed-off-by: yuexiao-wang <wang.yuexiao@zte.com.cn>

vdemeester committed:
Small cleanup in `integration-cli/docker_utils.go` 👼

- Move *one-shot* (one use) function where it is actually used (easier
  to know what's going on).
- Remove `pullImageIfNotExist` function as it might be an artifact
  from way back. We don't need it as we already have frozen/loaded
  image of busybox.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

omallo committed:
Reference new DigitalOcean plugin (Closes #29729)

Signed-off-by: Ovidio Mallo <ovidio.mallo@gmail.com>

coolljt0725 committed:
Follow up #29365, fix fail to remove container after restart

Call daemon.Mount will increase the refcount of mounted path,
for those previous running containers, `Mount` call will make
the refcount to 2. see
https://github.com/docker/docker/blob/v1.13.0-rc4/daemon/graphdriver/counter.go#L38
```
	if !m.check {
		m.check = true
		if c.checker.IsMounted(path) {
			m.count++
		}
	}
	m.count++

```
graphdrive could restore on reboot after #22541, call
daemon.Mount to resore the graphdriver is not necessary.

And call `daemon.Mount` on restorting will mount all the containers
mounted layer even if it was stop.

This fix call Mount and then Unmount to get `BaseFs`

Signed-off-by: Lei Jitang <leijitang@huawei.com>

Commit:
Fix flaky windows TestRestartRunningContainer test

I was seeing this for windowsRS1 testing:
17:20:36 ----------------------------------------------------------------------
17:20:36 FAIL: docker_cli_restart_test.go:31: DockerSuite.TestRestartRunningContainer
17:20:36
17:20:36 docker_cli_restart_test.go:39:
17:20:36     c.Assert(out, checker.Equals, "foobar\n")
17:20:36 ... obtained string = ""
17:20:36 ... expected string = "foobar\n"
17:20:36
17:20:59
17:20:59 ----------------------------------------------------------------------

and I think its because there's a delay between the time the container is
started and the 'echo' is actually run. This gives it up to 10 seconds
to do the 'echo' before giving up.

/cc @jhowardmsft

Signed-off-by: Doug Davis <dug@us.ibm.com>

allencloud committed:
fail fast when network filter invalid

Signed-off-by: allencloud <allen.sun@daocloud.io>

Commit:
test: Fix "--raw-logs=true" option test in TestDaemonStartWithoutColors

Because "tty" is closed in s.d.Stop() for "--raw-logs=false" test,
we need to open another pair of pty before test "--raw-logs=true"
option.

This patch fix #29772.

Signed-off-by: Jiuyue Ma <majiuyue@huawei.com>

Commit:
Complete the document content

Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

update

Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

allencloud committed:
update plugins in docs and correct things in index.md

Signed-off-by: allencloud <allen.sun@daocloud.io>

dnephin committed:
Add quoted string flag Value.

Signed-off-by: Daniel Nephin <dnephin@docker.com>

Commit:
Optimization a error description

Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

yuexiao-wang committed:
Update reference document for secret and stack

Signed-off-by: yuexiao-wang <wang.yuexiao@zte.com.cn>

vdemeester committed:
Remove pkg/integration and move it to testutil or integration-cli

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

vdemeester committed:
Add a registry package with registry v1/v2 code

This extract what was in registry_test.go and
registry_mock_test.go. This also move `RegistryHosting`
requirement to `registry.Hosting`

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

thaJeztah committed:
document DOCKER_HIDE_LEGACY_COMMANDS env-var

The `DOCKER_HIDE_LEGACY_COMMANDS` environment
variable was added in a7c8bcac2ba60d6dd25a1157085d9245bed556ce
but not documented.

This adds the variable to the documentation.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

albers committed:
Fix bash completion for `plugin enable|disable`

`docker plugin enable` and `docker plugin disable` only
accept one plugin.

Signed-off-by: Harald Albers <github@albersweb.de>

albers committed:
Add bash completion for `plugin disable --force`

Signed-off-by: Harald Albers <github@albersweb.de>

albers committed:
Fix usage message of `plugin inspect`

Signed-off-by: Harald Albers <github@albersweb.de>

jmzwcn committed:
API: Delete container can return 409, but that isn't documented. #29796

Signed-off-by: Daniel Zhang <jmzwcn@gmail.com>

vdemeester committed:
Remove deadcode from `service/opts.go`, `SecretOpt`

`SecretOpt` is in the `opts` package, this one is never used, so it's
dead code, removing it 👼.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

thaJeztah committed:
fix "retreive" typo in plugin store

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

vdemeester committed:
Updates some integration tests to fix userns builds

- `TestRunMountReadOnlyDevShm` and `TestRunHostnameInHostMode` needs
  `NotUserNamespace` requirement as these are known limitation of
  userns.
- `TestBuildWorkdirCmd` should use a preload image (`busybox`) instead
  of one that require network access.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

vdemeester committed:
Do not fail in `TearDown` if container not found when removing

If the container is not found when removing, it means it's already not
there anymore, so it's safe to ignore. This should reduce a bit some
`TearDown` flakyness..

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

Commit:
Remove the redundant 'if' statements

Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

Commit:
fix syscall.GUID composite literal uses unkeyed fields

Signed-off-by: yupengzte <yu.peng36@zte.com.cn>

vdemeester committed:
Should check for output, not error in deleteContainer

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

albers committed:
Support multiple services on bash completion for `docker service ps`

Signed-off-by: Harald Albers <github@albersweb.de>

Commit:
check both source_image_tag and target_image_tag for 'docker image tag'

Signed-off-by: wefine <wang.xiaoren@zte.com.cn>

thaJeztah committed:
bump go-units to e30f1e79f3cd72542f2026ceec18d3bd67ab859c

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

Commit:
Tidy kernel version in tests

Signed-off-by: John Howard <jhoward@microsoft.com>

dnephin committed:
Trim quotes from TLS flags.

Signed-off-by: Daniel Nephin <dnephin@docker.com>

Commit:
Tidy --isolation wording

Signed-off-by: John Howard <jhoward@microsoft.com>

vieux committed:
small secret doc update

Signed-off-by: Victor Vieux <vieux@docker.com>

thaJeztah commented on contrib/init/systemd/docker.service:
```diff
@@ -1,7 +1,7 @@
 Description=Docker Application Container Engine
 Documentation=https://docs.docker.com
-After=network.target docker.socket firewalld.service
+After=network-online.target docker.socket firewalld.service
```
Following the discussion in https://github.com/docker/docker/issues/29876#issue-198677228, @JamieGarside suggests to also add;

```
Wants=network-online.target
```

mentalMedley replied:
I tested this modification on my machine. It works as well as before. All containers autostarts correctly. Also I found [this about Wants](https://www.freedesktop.org/software/systemd/man/systemd.unit.html#Wants=):
> This is the recommended way to hook start-up of one unit to the start-up of another unit.

So Wants-option also must be added to Unit-file.

thaJeztah replied:
Can you update the PR with those changes? While you're doing that, can you squash your commits, so that there's a single commit in this PR? (some information on how to do that can be found in the [contributors guide](https://docs.docker.com/opensource/workflow/work-issue/#/pull-and-rebase-frequently))

mentalMedley replied:
I don't understand how to do git rebase. I have read contributors guide and @GordonTheTurtle 's advises and a lot of articles in Internet about it. I catch error "fatal: Needed a single revision
invalid upstream" and don't know what to do with this. It's too hard for me. Sorry :-( 
Please let @JamieGarside make his own pull request. I closed this PR.

thaJeztah commented:
ping @mavenugo @aaronlehmann PTAL as well

albers committed:
Add bash completion for plugin names

Signed-off-by: Harald Albers <github@albersweb.de>

Commit:
Correct comment in vol driver interface

Signed-off-by: John Howard <jhoward@microsoft.com>

cpuguy83 committed:
Fix race accessing plugin storage map

`plugins.GetAll()` was not locking the plugin map when checking if a
plugin exists, this can cause a race and potentially a panic if another
goroutine is trying to load a plugin into the map at the same time.

Also fixes a race during activation where a plugin inserts itself into
the plugin map but does not check if something else is already there.
This is already checked before trying to activate the plugin, however
the map lock is not held for this entire period, so other plugins may be
loaded during this time.
To fix, before inserting the plugin into the map, check if one with the
same name already exists and use that instead.

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

mavenugo committed:
Swarmkit has enabled the plugin filtering logic based on the drivers
returned in "docker info". Currently info endpoint isnt using the
GetAllByCap, but relies on existing networks to get the plugin names.
This causes a basic issue when it comes to global network plugins which
swarm-mode relies on, wherein swarmkit will not be able to schedule the
network on the worker nodes due to the filtering logic.

In order to break this chicken & egg issue, we must start to use the
GetAllManagedPluginsByCap. We are unable to use GetAllByCap due to
various issues with Plugin-V1's lazy loading approach causing issues
especially during daemon restarts (which uses SystemInfo)

Signed-off-by: Madhu Venugopal <madhu@docker.com>

Commit:
Windows: Enable TestBuildWorkdirImageCmd

Signed-off-by: John Howard <jhoward@microsoft.com>

Commit:
Windows: Fix old TP5 volume unit tests

Signed-off-by: John Howard <jhoward@microsoft.com>

YunxiangHuang committed:
Fix "zsh/_docker"

Signed-off-by: Yunxiang Huang <hyxqshk@vip.qq.com>

lixiaobing1 committed:
optimize Fatalf with the expectedError
Signed-off-by: lixiaobing10051267 <li.xiaobing1@zte.com.cn>

rhvgoyal committed:
devmapper: Return device id in error message

I often get complains that container removal failed and users got following
error message.

"Driver devicemapper failed to remove root filesystem 18a69ba82aaf7a039ce7d44156215012d703001643079775190ac7dd6c6acf56:Device is Busy"

This error message talks about container id but does not give any info
about which particular device id is busy. Most likely device is mounted
in some other mount namespace and if one knows the device id, they
can try to do some debugging figuring which process and which mount
namespace is keeping the device busy and how did we reach that stage.

Without that information, it becomes almost impossible to debug the
problem.

So to improve the debuggability, when device removal fails, also return
device id in error message. Now new message looks as follows.

"Driver devicemapper failed to remove root filesystem 18a69ba82aaf7a039ce7d44156215012d703001643079775190ac7dd6c6acf56: Failed to remove device dbc15bdf9994a17c613d8ef9e924f3cffbf67f91e4f709295c901ad628377991:Device is Busy"

Signed-off-by: Vivek Goyal <vgoyal@redhat.com>

jeveleth committed:
Fix minor typo

Signed-off-by: Josh Eveleth <joshe@opendns.com>

tonistiigi committed:
Set plugins used in tests back to original names

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

Commit:
vendor: Unfork logrus dependency

Vendor the latest official logrus release instead of a custom fork
(which now happens to be identical to the fork).

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

Commit:
Revendor swarmkit to 2e956c4

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

Commit:
Fix TestSecretCreateWithLabels nondeterminism

This test checks for the substring "foo" in the "secret ls" output. This
is a valid base36 substring and can sometimes show up by chance:

    docker_cli_secret_create_test.go:86:
        c.Assert(out, checker.Not(checker.Contains), name)
    ... obtained string = "" +
    ...     "ID                          NAME                        CREATED UPDATED\n" +
    ...     "ob8y4t4feuz8pn5h6vla9oxoz   stefoo7e268ozqfupi9s4se9q   Less than a second ago   Less than a second ago\n"
    ... substring string = "foo"

Change the secret name to test_secret, matching other tests. Underscores
can't appear in base36 so this name is safe to grep for.

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

lixiaobing1 committed:
Provide correct command imformation and URL

Signed-off-by: lixiaobing10051267 <li.xiaobing1@zte.com.cn>

yuexiao-wang committed:
Remove redundant return nil

Signed-off-by: yuexiao-wang <wang.yuexiao@zte.com.cn>

albers committed:
Fix wrong command in changelog

Signed-off-by: Harald Albers <github@albersweb.de>

tonistiigi committed:
Fix workdir cache invalidation

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

GordonTheTurtle commented:
<!-- AUTOMATED:POULE:DCO-EXPLANATION -->
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "fix_issue_18826" git@github.com:mentalMedley/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~842354625472
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Amending updates the existing PR. You **DO NOT** need to open a new one.

Dmitry-k42 committed:
fix issues #18826

Signed-off-by: Dmitry Kononenko <d@dm42.ru>

Dmitry-k42 committed:
Merge branch 'master' into fix_issue_18826

thaJeztah commented:
carrying this in https://github.com/docker/docker/issues/30210
